hw18-webflux (ДЗ: Комната 1408)

IlyaLeshin opened the pull request.

IlyaLeshin committed:
hw18-webflux (ДЗ: Комната 1408)
homework template

IlyaLeshin committed:
hw18-webflux (ДЗ: Комната 1408)
homework is done

IlyaLeshin committed:
hw18-webflux (ДЗ: Комната 1408)
homework is done with corrected constant in MessageController.java

petrelevich commented on hw18-webflux/client-service/src/main/java/ru/otus/controllers/MessageController.java:
```diff
@@ -0,0 +1,107 @@
+            saveMessage(roomId, message)
+                    .subscribe(msgId -> logger.info("message send id:{}", msgId));
+
+            template.convertAndSend(String.format("%s%s", TOPIC_TEMPLATE, roomId),
```
как выполняется эта часть ДЗ?

> Однако в нее приходят все сообщения из всех других комнат.

IlyaLeshin committed:
hw18-webflux (ДЗ: Комната 1408)
added a missing function(sending a message to room 1408 too) in MessageController.java

IlyaLeshin committed:
hw18-webflux (ДЗ: Комната 1408)
fix

petrelevich commented on hw18-webflux/client-service/src/main/java/ru/otus/controllers/MessageController.java:
```diff
@@ -0,0 +1,116 @@
+                });
+    }
+
+    private void convertAndSendMessageToThisAndSpecialRoom(String messageForSend, String thisRoomId, long specialRoomId) {
```
вместо этого сложного метода лучше два раза вызвать template.convertAndSend

IlyaLeshin committed:
hw18-webflux (ДЗ: Комната 1408)
fix

IlyaLeshin committed:
Merge branch 'main' into hw18-webflux
